coral/src/app/components/Form.tsx

```diff
@@ -145,6 +145,44 @@ export const TextInput = <T extends FieldValues>(
 
 TextInput.Skeleton = BaseInput.Skeleton;
 
+//
+// <NumberInput>
+//
+function _NumberInput<T extends FieldValues>({
+  name,
+  formContext: form,
+  ...props
+}: BaseInputProps & FormInputProps<T> & FormRegisterProps<T>) {
+  const { errors } = form.formState;
+  const error = get(errors, name)?.message as string;
+  return (
+    <BaseInput
+      {...props}
+      type="number"
+      {...form.register(name)}
+      valid={error ? false : undefined}
```

Comment:
This also looks weird to me, `valid` is always falsy? :o

Additional mega nitpick: usually am not a huge fan of implicit boolean checks like this. What is the falsy value we are checking against?

Reply:
Agreed! And now there is alternative suggestion for the evaluation.